apps/web/app/modules/Admin/EditCourse/CourseLessons/components/ChaptersList.tsx

```diff
@@ -0,0 +1,241 @@
+    async (event: React.MouseEvent) => {
+      event.stopPropagation();
+      try {
+        await updatePremiumStatus({
+          lessonId: chapter.id,
+          data: { isFree: !chapter.isFree },
+        });
+        queryClient.invalidateQueries({
+          queryKey: [COURSE_QUERY_KEY, { id }],
+        });
+      } catch (error) {
+        console.error("Failed to update chapter premium status:", error);
+      }
+    },
+    [chapter, updatePremiumStatus, id],
+  );
+
+  const lessonItems = useMemo(() => chapter.lessonItems, [chapter.lessonItems]);
+
+  return (
+    <AccordionItem
+      key={`chapter-${chapter.id}`}
+      value={`item-${chapter.id}`}
+      className="p-0"
+      draggable
+    >
+      <Card
+        className={cn("mb-4 h-full flex p-4 border", isOpen ? "border-[#5D84D4]" : "")}
+        onClick={onClickChapterCard}
+      >
+        <div className="flex w-full">
+          <div className="w-1/10 flex mt-2.5">
+            <Icon name="DragAndDropIcon" />
+          </div>
+          <div className="flex-1 flex flex-col justify-between ml-2">
+            <div className="flex justify-between items-center">
+              <div className="text-gray-600 text-sm">Chapter {chapter.displayOrder}</div>
+              <AccordionTrigger className="cursor-pointer" onClick={onAccordionClick}>
+                <Icon name={isOpen ? "ArrowUp" : "ArrowDown"} className="text-gray-600" />
+              </AccordionTrigger>
+            </div>
+            <h3 className="text-xl text-black">{chapter.title}</h3>
+            <AccordionContent className="mt-2 text-gray-700">
+              <div className="mt-4 grid grid-cols-1 gap-4">
+                {lessonItems.length === 0 ? (
+                  <p>No items for this lesson</p>
+                ) : (
+                  lessonItems.map((item) => {
+                    const key = item.content.id || `lesson-${item.content.id || Math.random()}`;
+                    return (
+                      <LessonCard key={key} item={item} onClickLessonCard={onClickLessonCard} />
+                    );
+                  })
+                )}
+              </div>
+            </AccordionContent>
+            <div className="mt-4 flex justify-between items-center">
+              <div className="flex items-center space-x-2">
+                <Button
+                  onClick={handleAddLessonClick}
+                  className="px-4 py-2 text-blue-600 bg-white border border-gray-300 rounded-md flex items-center"
+                >
+                  <Icon name="Plus" className="mr-2 text-blue-600" />
+                  Add Lesson
+                </Button>
+              </div>
+              <div className="flex items-center">
+                <Switch.Root
+                  className={cn(
+                    "w-10 h-6 mr-2 rounded-full relative transition-colors",
+                    chapter.isFree ? "bg-blue-500" : "bg-gray-200",
+                  )}
```

Comment:
What do you think about
```suggestion
                  className={cn(
                    "w-10 h-6 mr-2 rounded-full relative transition-colors",
                   {
                    "bg-blue-500": chapter.isFree,
                    "bg-gray-200": !chapter.isFree
                    }
                  )}
```